src/validations/access/on-chain/profiles.ts

```diff
@@ -1,113 +1,10 @@
-  async function sanitizeUrn(urn: string): Promise<string | undefined> {
-    if (!urn.startsWith('dcl://')) {
-      return urn
-    }
-    const parsed = await parseUrn(urn)
-    return parsed?.uri?.toString()
-  }
-
-  async function allOnChainWearableUrns(entity: Entity) {
-    const allWearablesInProfilePromises: Promise<string | undefined>[] = []
-    for (const avatar of entity.metadata.avatars) {
-      for (const wearableId of avatar.avatar.wearables) {
-        const isBaseAvatar = wearableId.includes('base-avatars')
-        if (!isBaseAvatar && !isOldEmote(wearableId)) {
-          allWearablesInProfilePromises.push(sanitizeUrn(wearableId))
-        }
-      }
-    }
-
-    return (await Promise.all(allWearablesInProfilePromises)).filter((wearableId): wearableId is string => !!wearableId)
-  }
-
-  async function allEmoteUrns(entity: Entity) {
-    const allEmotesInProfilePromises: Promise<string | undefined>[] = []
-    const allAvatars = entity.metadata?.avatars ?? []
-    for (const avatar of allAvatars) {
-      const allEmotes = avatar.avatar.emotes ?? []
-      for (const { urn } of allEmotes) {
-        if (!isOldEmote(urn)) {
-          allEmotesInProfilePromises.push(sanitizeUrn(urn))
-        }
-      }
-    }
-
-    return (await Promise.all(allEmotesInProfilePromises)).filter((wearableId): wearableId is string => !!wearableId)
-  }
-
-  return async function validateFn(deployment: DeploymentToValidate): Promise<ValidationResponse> {
-    const ethAddress = externalCalls.ownerAddress(deployment.auditInfo)
-    const depoymentTimestamp = deployment.entity.timestamp
-    const itemUrns: string[] = []
-    if (depoymentTimestamp >= ADR_75_TIMESTAMP) {
-      for (const urn of await allOnChainWearableUrns(deployment.entity)) {
-        itemUrns.push(urn)
-      }
-    }
-    if (depoymentTimestamp >= ADR_74_TIMESTAMP) {
-      for (const urn of await allEmoteUrns(deployment.entity)) {
-        itemUrns.push(urn)
-      }
-    }
-    const itemsOwnershipResult = await client.ownsItemsAtTimestamp(ethAddress, itemUrns, deployment.entity.timestamp)
-    if (!itemsOwnershipResult.result) {
-      return validationFailed(
-        `The following items (${itemsOwnershipResult.failing?.join(
-          ', '
-        )}) are not owned by the address ${ethAddress.toLowerCase()}).`
-      )
-    }
-
-    return OK
-  }
-}
+import { OnChainAccessCheckerComponents } from '../../../types'
+import { validateAll } from '../../validations'
+import { createItemOwnershipValidateFn, createNamesOwnershipValidateFn } from '../common/profile'
 
 export function createProfileValidateFn(components: Pick<OnChainAccessCheckerComponents, 'client' | 'externalCalls'>) {
   return validateAll(
-    createNamesOwnershipValidateFn(components),
-    createItemOwnershipValidateFn(components),
-    createPointerValidateFn(components)
+    createNamesOwnershipValidateFn(components, components.client),
```

Comment:
Shouldn't the receiver function get the client out of components?

Reply:
Yeah, but as this is a "common" validate fn, the component depends if the validator is TheGraph or OnChain